src/lang.ts

```diff
@@ -10,95 +10,55 @@ function isObject(item: any): boolean {
 	return item && typeof item === 'object' && !Array.isArray(item) && !(item instanceof RegExp);
 }
 
-function copyArray(array: any[], kwArgs: CopyArgs): any[] {
+function copyArray(array: any[], inherited: boolean): any[] {
 	return array.map(function (item: any): any {
 		if (Array.isArray(item)) {
-			return copyArray(item, kwArgs);
+			return copyArray(item, inherited);
 		}
 
-		return isObject(item) ?
-			copy({
+		return !isObject(item) ?
+			item :
+			_mixin({
+				deep: true,
+				inherited: inherited,
 				sources: [ item ],
-				deep: kwArgs.deep,
-				descriptors: kwArgs.descriptors,
-				inherited: kwArgs.inherited,
-				assignPrototype: kwArgs.assignPrototype
-			}) :
-			item;
+				target: Object.create(null)
+			});
 	});
 }
 
-export function copy(kwArgs: CopyArgs): any {
-	const sources: Hash<any>[] = kwArgs.sources;
-	let target: any;
-
-	if (!sources.length) {
-		throw new RangeError('lang.copy requires at least one source object.');
-	}
-
-	if (kwArgs.assignPrototype) {
-		// create from the same prototype
-		target = Object.create(Object.getPrototypeOf(sources[0]));
-	}
-	else {
-		// use the target or create a new object
-		target = kwArgs.target || {};
-	}
+interface MixinArgs {
+	deep: boolean;
+	inherited: boolean;
+	sources: {}[];
```

Comment:
This signature seems like an improvement over `CopyArgs` (which had `any`), but I'm wondering if there's anything better we can do.  AFAICT, this will match Arrays, Dates, and seemingly even primitives (though I honestly don't understand why).

I tried using `Hash<any>` in its place, but that makes passing valid object hashes more difficult in cases where implicit typing takes effect, e.g.:

```ts
assign(target, { foo: 42 }); // compiles

let obj = { foo: 42 };
assign(target, obj); // doesn't compile - needs explicit type to work
```

Any thoughts?  (cc @bryanforbes)